pkgs/by-name/li/libipuz/package.nix

```
@@ -1,44 +1,60 @@
 {
   lib,
   stdenv,
+  cargo,
   fetchFromGitLab,
+  gi-docgen,
+  glib,
+  gobject-introspection,
+  json-glib,
   meson,
   ninja,
   pkg-config,
-  glib,
-  json-glib,
+  rustPlatform,
 }:
-
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "libipuz";
-  version = "0.4.5";
+  version = "0.5.2";
 
   src = fetchFromGitLab {
     domain = "gitlab.gnome.org";
     owner = "jrb";
     repo = "libipuz";
-    rev = version;
-    hash = "sha256-psC2cFqSTlToCtCxwosXyJbmX/96AEI0xqzXtlc/HQE=";
+    rev = finalAttrs.version;
+    hash = "sha256-8bFMtqRD90SF9uT39Wkjf0eUef+0HgyrqY+DFA/xutI=";
+  };
+
+  cargoRoot = "libipuz/rust";
+  cargoDeps = rustPlatform.fetchCargoVendor {
+    inherit (finalAttrs) pname src version;
+    sourceRoot = "${finalAttrs.src.name}/${finalAttrs.cargoRoot}";
```

```suggestion
    inherit (finalAttrs) pname src version cargoRoot;
```

Can this work? Maybe you have to update hash